[clang][deps] Track VFS overlay files in file dependencies.

vsapsai opened the pull request:
rdar://164612831

vsapsai committed:
[clang][deps] Track VFS overlay files in file dependencies.

rdar://164612831

vsapsai commented:
Don't have a strong opinion on where tracking VFS files should be done. Mostly was trying to do that after `ModuleDepCollectorPP::handleTopLevelModule` prunes `VFSOverlayFiles` in `optimizeHeaderSearchOpts`.

jansvoboda11 commented on clang/lib/Tooling/DependencyScanning/ModuleDepCollector.cpp:
```diff
@@ -616,6 +616,10 @@ void ModuleDepCollectorPP::EndOfMainFile() {
   if (!MDC.ScanInstance.getPreprocessorOpts().ImplicitPCHInclude.empty())
     MDC.addFileDep(MDC.ScanInstance.getPreprocessorOpts().ImplicitPCHInclude);
 
+  for (const std::string &Overlay :
```
```suggestion
  for (StringRef Overlay :
```

jansvoboda11 reviewed (approved):
LGTM!

vsapsai committed:
Change type per review request. Change variable name to fit on a line.

It is OK to use `StringRef` because the method type is `void ModuleDepCollector::addFileDep(StringRef Path)`
so we aren't adding extra unnecessary `StringRef` instances.
